### ISSUE TITLE
Consider whether service fields can represent session state

### ISSUE THREAD

**Issue by jclark** [ballerina-platform/ballerina-spec#290]
The spec allows services to have only methods. It would be nice if we could make it work for service fields to represent session state. (This would imply adding an __init method #289.)

The difficulty is in making this work sensibly in a transport-agnostic way.

**Comment by sanjiva** [comment]
@hasithaa in #289 you have said we allow fields in services right now. Is that correct?? I don't see how it can work as that requires associating "service instances" to some type of session.

We had a session concept (for HTTP) but that was a layer above this IIRC.

**Comment by hasithaa** [comment]
@sanjiva, Yes, it was,  In the implementation, this works because we use objects to model services. A service constructor is converted into an anonymous object type and then create a value (newing) of that type. This gave us a service instance to use as a session. 

IIRC, In one of the iterations, we decided to disallow fields inside a service. Then we brought it back, because, those can be treated as global variables lexically scoped to a service. But with the introduction of service constructor, it is no longer valid, (because service constructors can be inside a function), But we missed this change during that time. 

Btw, We fixed this issue in master (Fields in services are disallowed now).
